linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -3097,27 +3219,39 @@ private object FunctionEmitter {
     case object Discard extends Lhs
   }
 
+  sealed abstract class VarKind
+
+  object VarKind {
+    case object Mutable extends VarKind
+    case object Immutable extends VarKind
+    case object ThisAlias extends VarKind
+    case object ClassCapture extends VarKind
+
+    /** Alias to a file level (!) ident. */
+    final case class VarAlias(ident: js.Ident) extends VarKind
```

Comment:
Alternatively, we could just add `ExplicitThisAlias` here and change `thisIdent` in the env to `hasExplicitThis: Boolean`. (in practice, `thisIdent` is always `None` or `Some(fileLevelVarIdent("thiz", thisOriginalName))`).

This would be safer.

Reply:
It seems fine enough to me as is. Perhaps call it `FileLevelVarAlias` if that is indeed the requirement for its usage to be fine?

Reply:
I have added a commit showing the alternative with `ExplicitThisAlias`. IMO its less complicated. Notably, there are less perceived degrees of freedom.